### PR TITLE
2025 - Menu button now expands to show vertical navbar

### PR DESCRIPTION
**Pull request by kpherwan**
Issue - https://github.com/informatics-isi-edu/chaise/issues/2025

**Before -** 
Clicking on Menu expands nothing

image

**After -**
Navbar visible

image

**Test link -** 
https://dev.rebuildingakidney.org/~kpherwan/chaise/recordset/#2/Microarray:Series@sort(RID)

https://dev.facebase.org/chaise/recordset/#1/isa:dataset@sort(release_date::desc::,RID)

### CODE REVIEW

File: common/styles/scss/_navbar.scss [hunk 1]
```diff
@@ -173,3 +173,7 @@ body > .container {
     overflow-x: hidden;
     overflow-y: auto;
 }
+
+.navbar-collapse {
+    max-height: 340px;
```
Comment (RFSH) [inline common/styles/scss/_navbar.scss:178]: Why this specific value? Is this based on RBK's navbar?

Comment (kpherwan) [reply common/styles/scss/_navbar.scss:178]: This value is in bootstrap for navbar-collapse but it isn't getting applied so I have added it here

**Comment by RFSH** [comment]
The changes look good. Before merging, 

- Test this collapsed version with the default CSS rules hat we have in chaise (don't include any custom css like RBK and Facebase are doing.) And make sure the default UI looks good. For example the default background-color applied to `.child-opened` by bootstrap doesn't work with our navbar (We have a black navbar, so this one most probably should be some shades of grey).

- Test this in RBK and Facebase. It should work without any issues, the reason for testing is making sure the UI looks good. As you can see in different deployments, we're using different CSS rules to customize the navbar and the colors might be off when the navbar is collapsed. You don't need to make any CSS changes in different deployments, just testing it and noting what you think should be changed is enough.

- I think it's good to document the useful css classes and HTML structure of navbar to make the process of customizing navbar easier. We have a document for navbar that you can find [here](https://github.com/informatics-isi-edu/chaise/blob/master/docs/user-docs/navbar-app.md). Add a "Customize navbar UI" section before "Notes". You should,

   - Add a minimal representation of HTML structure in there.
   - Explain what are the common classes and their purpose. For example you can mention `navbar-collapse` and explain that the height for the collapsed version is set in there. You don't need to explain every class in there. Just what seems more relevant (How to change color, how to change height, etc)

File: common/styles/scss/_navbar.scss (hunk 1 above)
Comment (kpherwan) [reply common/styles/scss/_navbar.scss:179]: Done

**Comment by kpherwan** [comment]
> As Aref mentioned, the default background-color applied to .child-opened doesn't work with our navbar in the vertical setting, changed to show submenu items indented to indicate the expansion.

image

> Added the documentation for "Customize navbar UI" to make it easier for users to find what they would need to change -
https://github.com/informatics-isi-edu/chaise/blob/2025-responsive-menu-bug/docs/user-docs/navbar-app.md

**Review by RFSH — APPROVED** [review]

**Pull request merged**